### PR TITLE
Initialized Member Variables in /modules

### PR DESCRIPTION
**Pull request by webbuf**
Initializing class member variables in /modules for #52674

### CODE REVIEW

**Comment by fire** [comment]
From my understand, Godot Engine code tends to initialize in the class member and not via member initialization list.

File: modules/visual_script/editor/visual_script_property_selector.cpp [hunk 1]
```diff
@@ -379,6 +379,9 @@ void VisualScriptPropertySelector::_bind_methods() {
 }
 
 VisualScriptPropertySelector::VisualScriptPropertySelector() {
+	properties = false;
+	visual_script_generic = false;
+	connecting = false;
 	virtuals_only = false;
```
Comment (akien-mga) [inline modules/visual_script/editor/visual_script_property_selector.cpp:385]: These should be initialized in the header (definition).

File: modules/visual_script/editor/visual_script_editor.h [hunk 2]
```diff
@@ -196,7 +196,7 @@ class VisualScriptEditor : public ScriptEditorBase {
 	int _create_new_node_from_name(const String &p_text, const Vector2 &p_point);
 	void _selected_new_virtual_method(const String &p_text, const String &p_category, const bool p_connecting);
 
-	int error_line;
+	int error_line = 0;
```
Comment (akien-mga) [inline modules/visual_script/editor/visual_script_editor.h:199]: Based on the code using it in the `.cpp`, it seems -1 would be a better default value (that's what `set_debugger_active` resets it to).

File: modules/mono/csharp_script.h [hunk 3]
```diff
@@ -103,7 +103,7 @@ class CSharpScript : public Script {
 	bool valid = false;
 	bool reload_invalidated = false;
 
-	bool builtin;
+	bool builtin = false;
```
Comment (akien-mga) [inline modules/mono/csharp_script.h:106]: This actually seems unused, so it should be removed. No reference to it in the `.cpp`, and it's private.

File: modules/basis_universal/texture_basisu.cpp [hunk 4]
```diff
@@ -208,6 +208,7 @@ Vector<uint8_t> TextureBasisU::get_basisu_data() const {
 
 TextureBasisU::TextureBasisU() {
 	texture = RenderingServer::get_singleton()->texture_create();
+	tex_size = Size2(0, 0);
```
Comment (akien-mga) [inline modules/basis_universal/texture_basisu.cpp:211]: This is not necessary as `Size2 tex_size;` is already default initialized.

File: modules/bullet/shape_bullet.cpp [hunk 5]
```diff
@@ -163,7 +163,7 @@ btRayShape *ShapeBullet::create_shape_ray(real_t p_length, bool p_slips_on_slope
 /* World boundary */
 
 WorldBoundaryShapeBullet::WorldBoundaryShapeBullet() :
-		ShapeBullet() {}
+		ShapeBullet(), plane(Plane()) {}
```
Comment (akien-mga) [inline modules/bullet/shape_bullet.cpp:166]: `Plane` is default initialized so this isn't needed.

File: modules/bullet/shape_bullet.cpp [hunk 6]
```diff
@@ -244,7 +244,7 @@ btCollisionShape *BoxShapeBullet::create_bt_shape(const btVector3 &p_implicit_sc
 /* Capsule */
 
 CapsuleShapeBullet::CapsuleShapeBullet() :
-		ShapeBullet() {}
+		ShapeBullet(), height(0.0), radius(0.0) {}
```
Comment (akien-mga) [inline modules/bullet/shape_bullet.cpp:247]: These should be set in the header (same for other shapes).

**Comment by akien-mga** [comment]
To do changes to this PR based on feedback, please use the workflow described here (`git commit --amend` and `git push --force`): https://docs.godotengine.org/en/latest/community/contributing/pr_workflow.html#modifying-a-pull-request

Additionally, the email you configured for your Git author information is not a valid email, and not linked to your GitHub account. See the header of https://patch-diff.githubusercontent.com/raw/godotengine/godot/pull/57467.patch

You can fix it by changing the email in your Git config and re-authoring the commit with `git commit --amend --reset-author`. If you want the commit to be linked to your GitHub account (so you can be credited for it), you should use an email that is known to your GitHub settings.

File: modules/bullet/shape_bullet.cpp [hunk 7]
```diff
@@ -191,7 +191,7 @@ btCollisionShape *WorldBoundaryShapeBullet::create_bt_shape(const btVector3 &p_i
 /* Sphere */
 
 SphereShapeBullet::SphereShapeBullet() :
-		ShapeBullet() {}
+		ShapeBullet(), radius(0.0) {}
```
Comment (akien-mga) [inline modules/bullet/shape_bullet.cpp:194]: radius should also be initialized in the header.

**Review by akien-mga — APPROVED** [review]
Last nitpick, otherwise looks good.

**Comment by fire** [comment]
Approved cicd.

**Comment by akien-mga** [comment]
Thanks! And congrats for your first merged Godot contribution :tada:

**Pull request merged**